### PR TITLE
supports auto route homa

### PR DESCRIPTION
**Pull request by shunjizhan**
## Context
Currently UI needs to wait for token arrival at router address, then call relayer `route`. But there are many scenarios that breaks the process, and the fund is stuck at the router address.

To solve this issue, we can upgrade the relayer to be able to "wait and route", so the whole process is more robust. Now UI only needs to send one request to the relayer right after user signed xcm tx, and don't need to worry about user closing UI in the middle anymore.

This will also enhance UX by displaying a progress bar: `xcm (waiting) => routing => confirming => complete`

## Change
- separated different route apis from `api/route.ts` to specific files to be more clean
- added 3 new endpoints:
  - `/routeHomaAuto`: wait for token arrive then auto perform the route, returns a route Id so frontend can use this to track and display progress. 
  - `/routeStatus`: get current route status
  - `/allRouteStatus`: see all route status, mostly for debugging purpose

Note that we don't need to permanently store route status, as it's only for frontend to track progress and display progress bar. For a complete route history, we can add new endpoint with subquery in the future

## Test
- added tests for auto route homa flow, tested locally and worked
- CI has been broken for a while, will fix it before releasing stable version

### CODE REVIEW

File: src/api/homa.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  Timeout = -1,
+  Failed = -2,
+};
+
+interface RouteInfo {
+  status: RouteStatus;
+  [key: string]: any;
+}
+
+let routeReqId = 0;
```
Comment (xlc) [inline src/api/homa.ts:72]: this means id will reset after server restart.
how about just make it start from `Date.now()`

Comment (shunjizhan) [reply src/api/homa.ts:72]: yeah that's a good idea, I will make `const reqId = `homa-${Date.now()}`;` so each id is unique and contains info about when it is received

Comment (shunjizhan) [reply src/api/homa.ts:72]: currently doing a couple small polishes, will commit the changes together

Comment (xlc) [reply src/api/homa.ts:72]: only using timestamp is not good because it could have collections. so either add another sequential number or a random string

Comment (shunjizhan) [reply src/api/homa.ts]: ok let's make seqId starts from Date.now()

**Review by xlc — APPROVED** [review]

**Pull request merged**